test/powershell/Modules/Microsoft.PowerShell.Security/ConstrainedLanguageRestriction.Tests.ps1

```diff
@@ -0,0 +1,890 @@
+        if (-not (Test-Path $moduleDirectory))
+        {
+            mkdir $moduleDirectory -Force | Out-Null
+        }
+
+        try 
+        { 
+            Add-Type -TypeDefinition $code -OutputAssembly $moduleDirectory\TestCmdletForConstrainedLanguage.dll -ErrorAction Ignore 
+        } catch {}
+
+        Import-Module -Name $moduleDirectory\TestCmdletForConstrainedLanguage.dll
+    }
+} # end if ($IsWindows)
+
+try
+{
+    $defaultParamValues = $PSDefaultParameterValues
+    $PSDefaultParameterValues["it:Skip"] = !$IsWindows
+
+    Describe "Built-ins work within constrained language" -Tags 'Feature' {
+
+        BeforeAll {
+            $TestCasesBuiltIn = @(
+                @{testName = "Verify built-in function"; scriptblock = { Get-Verb } }
+                @{testName = "Verify build-in error variable"; scriptblock = { Write-Error SomeError -ErrorVariable ErrorOutput -ErrorAction SilentlyContinue; $ErrorOutput } }
+            )
+        }
+
+        It "<testName>" -TestCases $TestCasesBuiltIn {
+
+            param ($scriptblock)
+
+            $ExecutionContext.SessionState.LanguageMode = "ConstrainedLanguage"
+            $result = (& $scriptblock)
+            Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguageMode
+
+            ($result.Count -gt 0) | Should Be $true
+        }
+
+        AfterEach {
+            Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguageMode
+        }
+    }
+
+    Describe "Background jobs" -Tags 'Feature' {
+
+        Context "Background jobs in system lock down mode" {
+
+            # TODO
+            # Test is pending because PowerShell Core currently stubs out Windows system lockdown APIs
+            It "Verifies that background jobs in system lockdown mode run in constrained language" -Pending {
+
+                Invoke-LanguageModeTestingSupportCmdlet -SetLockdownMode
+
+                $job = Start-Job -ScriptBlock { [object]::Equals("A", "B") } | Wait-Job
+                $expectedError = $job.ChildJobs[0].Error
+                $job | Remove-Job
+
+                Invoke-LanguageModeTestingSupportCmdlet -RevertLockdownMode
+
+                $expectedError | Should Not Be $null
+                $expectedError.FullyQualifiedErrorId | Should Match "MethodInvocationNotSupportedInConstrainedLanguage"
+            }
+        }
+
+        Context "Background jobs within inconsistent mode" {
+
+            It "Verifies that background job is denied when mode is inconsistent" {
+
+                $ExecutionContext.SessionState.LanguageMode = "ConstrainedLanguage"
+
+                try
+                {
+                    Start-Job { [object]::Equals("A", "B") }
+                    throw "No Exception!"
+                }
+                catch 
+                {
+                    $exception = $_
+                }
+
+                Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguageMode
+

```

Comment:
the above block should probably be:
```
try
{
    Start-Job { [object]::Equals("A","B") }
    throw "No Exception!"
}
catch
{
    $exception = $_
}
finally
{
    Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguageMode
}
```
and the same below

Reply:
Agreed.  I was trying to use Pester AfterAll, AfterEach.  But I prefer try/finally.